src/pages/studyView/virtualStudy/VirtualStudy.tsx

```diff
@@ -0,0 +1,248 @@
+                        />
+                        {this.props.name}
+                    </span>
+                    <a target="_blank" href={`newstudy?id=${this.props.studyId}`}>
+                        <i className="fa fa-external-link" aria-hidden="true"></i>
+                    </a>
+                </div>
+                <div className={styles.studyDescription} style={{ display: this.showDescription ? 'block' : 'none' }}>
+                    <span dangerouslySetInnerHTML={{ __html: `${this.props.description.replace(/\r?\n/g, '<br/>')}` }} />
+                </div>
+            </div>
+        )
+    }
+}
+
+@observer
+export default class VirtualStudy extends React.Component<IVirtualStudyProps, {}> {
+
+    @observable private name: string = '';
+    @observable private description: string = '';
+
+    @observable private saving = false;
+    @observable private sharing = false;
+    @observable private copied = false;
+
+    constructor(props: IVirtualStudyProps) {
+        super(props);
+        this.description = getVirtualStudyDescription(props.studyWithSamples, props.selectedSamples, props.filter, this.props.attributeNamesSet, this.props.user);
+    }
+
+    @computed get namePlaceHolder() {
+        return `Selected sample${this.props.selectedSamples.length > 1 ? 's' : ''} (${getCurrentDate()})`
+    }
+
+    @computed get buttonsDisabled() {
+        return _.isEmpty(this.namePlaceHolder) && _.isEmpty(this.name);
+    }
+
+    readonly virtualStudy = remoteData({
+        invoke: async () => {
+            if (this.saving || this.sharing) {
+                let selectedSampleSet = _.groupBy(this.props.selectedSamples, (sample: Sample) => sample.studyId);
+                let studies = _.reduce(selectedSampleSet, (acc: { id: string; samples: string[] }[], samples, studyId) => {
+                    acc.push({
+                        id: studyId,
+                        samples: samples.map(sample => sample.sampleId)
+                    })
+                    return acc;
+                }, []);
+
+                let filters = { patients: {}, samples: {} };
+
+                /* 
+                    TODO: this is to support existing virtual study feature.
+                    but eventually we need to save StudyViewFilter
+                 */
+                let parameters = {
+                    name: this.name || this.namePlaceHolder,
+                    description: this.description,
+                    filters: filters,
+                    origin: this.props.studyWithSamples.map(study => study.studyId),
+                    studies: studies
+                }
+                return await sessionServiceClient.saveVirtualStudy(parameters, this.saving);
+            }
+            return undefined;
+        }
+    }, undefined);
+
+    @computed get virtualStudyUrl() {
+        return getStudySummaryUrl(this.virtualStudy.result ? this.virtualStudy.result.id : '');
+    }
+
+    @autobind
```

Comment:
What's the difference using `@autoband` and `@bind`

Reply:
both purpose the same functionality, but different libraries. From one of the planning meeting I remember we decided going with `@autobind`